### PR TITLE
Profiler improvements

### PR DESCRIPTION
**Pull request by aztech-dev**
Adds a new WebProfilerServiceProvider that provides the following services:

- Ajax profiling tab (not available in Silex provider for v1.0)
- Custom Doctrine profiler init (other service provider does play along nice)
- Cache profiling tab

### CODE REVIEW

File: lib/Alchemy/Phrasea/Core/Provider/WebProfilerServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            $cache->setNamespace($namespace);
+
+            return $cache;
+        }));
+
+        $app['data_collector.cache_subscriber'] = $app->share(function ($app) {
+            return new CacheStatisticsSubscriber($app['cache']);
+        });
+
+        $app->share($app->extend('data_collectors', function ($collectors) use ($app) {
```
Comment (bburnichon) [inline lib/Alchemy/Phrasea/Core/Provider/WebProfilerServiceProvider.php]: `$app['data_collectors'] = $app->share($app->extends('data_collectors', ...));`

File: lib/Alchemy/Phrasea/Core/Provider/WebProfilerServiceProvider.php [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            $collectors['db'] = $app->share(function ($app) {
```
Comment (bburnichon) [inline lib/Alchemy/Phrasea/Core/Provider/WebProfilerServiceProvider.php]: That's `$collectors` you are extending, so `$collectors['db'] = $collectors->share(function () use ($app) {`

Comment (aztech-dev) [inline lib/Alchemy/Phrasea/Core/Provider/WebProfilerServiceProvider.php]: $collectors is an array, not a Pimple container...

File: lib/Alchemy/Phrasea/Core/Provider/WebProfilerServiceProvider.php (hunk 1 above)
Comment (aztech-dev) [inline lib/Alchemy/Phrasea/Core/Provider/WebProfilerServiceProvider.php]: $collectors is an array, not a Pimple container...

File: templates-profiler/cache.html.twig [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                <th>Key</th>
+                <th>Hit / Miss</th>
+            </tr>
+        </thead>
+        <tbody>
+        {% for call in collector.calls %}
+            <tr>
+                <td>{{ loop.index }}</td>
+                <td>
+                    {% if call['type'] != 'fetch' %}<strong>{% endif %}
```
Comment (bburnichon) [inline templates-profiler/cache.html.twig]:
```twig
{% if call['type'] != 'fetch' %}
  <strong>{{ call['type'] }}</strong>
{% else %}
  {{ call['type'] }}
{% endif %}
```

File: lib/Alchemy/Phrasea/Core/Provider/WebProfilerServiceProvider.php (hunk 1 above)
Comment (bburnichon) [inline lib/Alchemy/Phrasea/Core/Provider/WebProfilerServiceProvider.php]: oups, false guess...

You should still assign `$app->share()` to `$app['data_collectors']`

File: lib/Alchemy/Phrasea/Core/Provider/WebProfilerServiceProvider.php (hunk 2 above)
Comment (bburnichon) [inline lib/Alchemy/Phrasea/Core/Provider/WebProfilerServiceProvider.php]: Missing assignment

**Pull request merged**